src/character/mod.rs

```diff
@@ -48,11 +48,16 @@ fn default_class() -> &'static Class {
 
 impl Character {
     pub fn player() -> Self {
-        Self::new(&Class::HERO, 1)
+        Self::new(&Class::HERO, 1, None, None)
     }
 
     pub fn enemy(level: i32, distance: location::Distance) -> Self {
-        Self::new(Class::random_enemy(distance), level)
+        Self::new(Class::random_enemy(distance), level, None, None)
+    }
+
+    pub fn ascend(class: &'static Class, level: i32, sword: Option<equipment::Sword>,
+                   shield: Option<equipment::Shield>) -> Self {
+        Self::new(class, level, sword, shield)
```

Comment:
can't you just do `self.class = new_class`  here instead creating a new one (and having to pass sword/shield)?